src/omotes_simulator_core/adapter/transforms/esdl_asset_mapper.py

```diff
@@ -60,11 +65,25 @@ def to_entity(model: EsdlAssetObject) -> AssetAbstract:
 
         :return: Entity object of type AssetAbstract.
         """
-        if not type(model.esdl_asset) in CONVERSION_DICT:
+        if (
+            not type(model.esdl_asset) in CONVERSION_DICT
+            and not type(model.esdl_asset) in conversion_dict_mappers
+        ):
             raise NotImplementedError(str(model.esdl_asset) + " not implemented in conversion")
-        return CONVERSION_DICT[type(model.esdl_asset)](
-            model.esdl_asset.name, model.esdl_asset.id, model.get_port_ids()
+
+        # Use the dictionary to get the appropriate mapper
+        asset_type = type(model.esdl_asset)
+        if asset_type in conversion_dict_mappers:
+            mapper = conversion_dict_mappers[asset_type]()
+            return mapper.to_entity(model)
+
+        converted_asset = CONVERSION_DICT[type(model.esdl_asset)](
```

Comment:
The two seperate if loops will be removed when all assets have a mapper, right?

Reply:
yes.